rules/catch-error-name.js

```diff
@@ -34,7 +34,7 @@ const create = context => {
 
 	const options = {
 		name: 'error',
-		caughtErrorsIgnorePattern: '^_$',
+		caughtErrorsIgnorePattern: /^_$|^error\w+|^[A-Za-z]\w*(e|E)rror$/.source,
```

Comment:
Should we really allow `errorFoo`? Seems like the more correct naming would be `fooError`.

Reply:
Are you asking "why don't we support regex **literal** directly"?

Reply:
Sorry, yes. I mean we can support regex literal and regex string.

Reply:
How? The JSON schema does not support regex literal.

Reply:
Right,never mind.

Reply:
Relevant: https://github.com/eslint/eslint/issues/11318

Reply:
Removed support for `errorFoo`.